x-pack/plugins/endpoint/public/embeddables/resolver/store/middleware.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreStart } from 'kibana/public';
+import { Dispatch, MiddlewareAPI } from 'redux';
+import { ResolverState, ResolverAction } from '../types';
+
+type MiddlewareFactory<S = ResolverState> = (
+  coreStart: CoreStart
+) => (
+  api: MiddlewareAPI<Dispatch<ResolverAction>, S>
+) => (next: Dispatch<ResolverAction>) => (action: ResolverAction) => unknown;
+
+export const resolverMiddlewareFactory: MiddlewareFactory = coreStart => {
+  return api => next => async (action: ResolverAction) => {
+    next(action);
+    if (action.type === 'userChangedSelectedEvent') {
+      // this is brittle as it is going to change assumedly.
+      api.dispatch({ type: 'appRequestedResolverData' });
+      const {
+        coreStart: { http },
+      } = coreStart;
+      //const uniquePid = action.payload.selectedEvent?.value?.source.endgame.data.pid;
```

Comment:
TODO: use the correct values from the event a user selects, decide if we want/need to support both v1 and v0 data formats.